react-native-ts/src/ItemListView.tsx

```diff
@@ -0,0 +1,156 @@
+import React, {useCallback, useState, useEffect} from 'react';
+import {BSON} from 'realm';
+import {useUser} from '@realm/react';
+import {SafeAreaProvider} from 'react-native-safe-area-context';
+import {StyleSheet, Text, View} from 'react-native';
+import {Button, Overlay, ListItem} from 'react-native-elements';
+import Icon from 'react-native-vector-icons/FontAwesome';
+
+import {CreateToDoPrompt} from './CreateToDoPrompt';
+import realmContext from './realmContext';
+import {Item} from './ItemSchema';
+const {useRealm, useQuery} = realmContext;
+
+Icon.loadFont(); // load FontAwesome font
+
+export function ItemListView() {
+  const realm = useRealm();
+  const items = useQuery(Item);
+  const user = useUser();
+  const [showNewItemOverlay, setShowNewItemOverlay] = useState(false);
+
+  useEffect(() => {
+    // initialize the subscriptions
+    const updateSubscriptions = async () => {
```

Comment:
Does this effect need a cleanup or is it fire and forget? i.e. what happens if the ItemListView unmounts before realm.subscriptions.update() completion? If it's fine, a comment might help. Do subscriptions need to be cleaned up? (`mutableSubs.remove(...)`)

Reply:
Good point, this should return a function that removes the subscription.  At some point I want to introduce a hook called `useSubscription` that automatically does this.